### PR TITLE
Read database config from Flytestdlib

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
Failed to register database section in single binary because we've registered it in datacatalog.
Here we read config from data catalog if it exists

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2266

## Follow-up issue
https://github.com/flyteorg/flytestdlib/pull/121

### CODE REVIEW

File: pkg/runtime/application_config_provider.go [hunk 1]
```diff
@@ -90,6 +91,47 @@ var externalEventsConfig = config.MustRegisterSection(externalEvents, &interface
 type ApplicationConfigurationProvider struct{}
 
 func (p *ApplicationConfigurationProvider) GetDbConfig() *interfaces.DbConfig {
+	// The error only occurs when using single binary to launch flyte cluster.
```
Comment (EngHabu) [inline pkg/runtime/application_config_provider.go:94]: Should this entire package be moved somewhere common? Stdlib or something? Having two packages register the same config section isn't a good pattern... 

Maybe just the db connection part?

Comment (pingsutw) [reply pkg/runtime/application_config_provider.go:94]: yeah, agree, I think we should move to `flytestdlib`.

File: pkg/runtime/application_config_provider.go [hunk 2]
```diff
@@ -90,7 +77,17 @@ var externalEventsConfig = config.MustRegisterSection(externalEvents, &interface
...
+	databaseConfig := database.GetConfig()
+	switch {
+	case databaseConfig.SQLite != database.SQLiteConfig{}:
+		sqliteConfig := interfaces.SQLiteConfig(databaseConfig.SQLite)
+		return &interfaces.DbConfig{SQLiteConfig: &sqliteConfig}
+	case databaseConfig.Postgres != database.PostgresConfig{}:
+		postgresConfig := interfaces.PostgresConfig(databaseConfig.Postgres)
+		return &interfaces.DbConfig{PostgresConfig: &postgresConfig}
+	default:
+		return &interfaces.DbConfig{}
```
Comment (katrogan) [inline pkg/runtime/application_config_provider.go:89]: should we log in this case?

**Review by katrogan — APPROVED** [review]

File: pkg/runtime/application_config_provider.go [hunk 3]
```diff
@@ -90,7 +77,17 @@ var externalEventsConfig = config.MustRegisterSection(externalEvents, &interface
 type ApplicationConfigurationProvider struct{}
 
 func (p *ApplicationConfigurationProvider) GetDbConfig() *interfaces.DbConfig {
-	return databaseConfig.GetConfig().(*interfaces.DbConfig)
+	databaseConfig := database.GetConfig()
+	switch {
+	case databaseConfig.SQLite != database.SQLiteConfig{}:
```
Comment (EngHabu) [inline pkg/runtime/application_config_provider.go:82]:
```suggestion
	case !databaseConfig.SQLite.IsEmpty():
```

File: pkg/runtime/application_config_provider.go (hunk 2 above)
Comment (EngHabu) [reply pkg/runtime/application_config_provider.go:89]: We should probably error?

File: go.mod [hunk 4]
```diff
@@ -172,3 +175,5 @@ require (
 )
 
 replace github.com/robfig/cron/v3 => github.com/unionai/cron/v3 v3.0.2-0.20210825070134-bfc34418fe84
+
+replace github.com/flyteorg/datacatalog => github.com/pingsutw/datacatalog v0.3.20-0.20220327153301-99505e14d76a
```
Comment (EngHabu) [inline go.mod:179]: delete?

Comment (pingsutw) [reply go.mod]: oops, sorry my bad

**Review by EngHabu — APPROVED** [review]

**Comment by EngHabu** [comment]
image

**Review by EngHabu — APPROVED** [review]
